### PR TITLE
[PREVIEW] Non-repudiation config

### PR DESCRIPTION
**Pull request by luigibk**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RPE-678

### Change description ###
Non-repudiation configuration

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: infrastructure/main.tf [hunk 1]
```diff
@@ -100,6 +100,9 @@ module "bulk-scan" {
     SCAN_ENABLED                  = "${var.scan_enabled}"
     STORAGE_BLOB_LEASE_TIMEOUT    = "${var.blob_lease_timeout}" // In seconds
 
+    STORAGE_BLOB_SIGNATURE_ALGORITHM = "none" // none or sha256withrsa
+    STORAGE_BLOB_PUBLIC_KEY          = "${data.azurerm_key_vault_secret.exela-public-key.value}"
```
Comment (rkondratowicz) [inline infrastructure/main.tf]: should exela name be here?

Comment (luigibk) [reply infrastructure/main.tf]: It would be actually better if it was clear whose public key that is. @njrich28 is this an issue?

Comment (doncem) [reply infrastructure/main.tf]: it was initially. rafal's question is valid

Comment (KevinJCross) [reply infrastructure/main.tf]: true ... function rather than company name would be a better fit :)

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/TestHelper.java [hunk 2]
```diff
@@ -35,6 +39,8 @@
 
 public class TestHelper {
 
+    public static final String SIGNATURE_FILE = "test_private_key.der";
```
Comment (rkondratowicz) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/TestHelper.java]: perhaps `SIGNING_KEY_FILE`?

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/ZipVerifiers.java [hunk 3]
```diff
@@ -148,6 +186,26 @@ public ZipStreamWithSignature(
...
+            String zipFileName,
+            String container
+        ) {
+            String publicKeyBase64 = null;
+            try {
+                if (!Strings.isNullOrEmpty(publicKeyDerFile) && !"none".equalsIgnoreCase(publicKeyDerFile)) {
+                    publicKeyBase64 =
+                        Base64.getEncoder().encodeToString(
+                            Resources.toByteArray(getResource(publicKeyDerFile))
+                        );
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/ZipVerifiers.java]: It would be nice to read this file only once, on application start.
Happy to do have it done on a separate PR.
